src/main/java/fr/xephi/authme/listener/PlayerListener.java

```diff
@@ -481,58 +446,67 @@ public void onSignChange(SignChangeEvent event) {
-            PlayerAuth auth = PlayerAuth.builder()
-                .name(name)
-                .realName(player.getName())
-                .location(spawn)
-                .build();
-            dataSource.updateQuitLoc(auth);
-            bungeeSender.sendAuthMeBungeecordMessage(MessageType.REFRESH_QUITLOC, name);
-        }
-        if (spawn != null && spawn.getWorld() != null) {
-            event.setRespawnLocation(spawn);
+    /*
+     * Inventory interactions
+     */
+
+    @EventHandler(ignoreCancelled = true, priority = EventPriority.LOWEST)
+    public void onPlayerPickupItem(PlayerPickupItemEvent event) {
+        if (listenerService.shouldCancelEvent(event)) {
+            event.setCancelled(true);
         }
     }
 
     @EventHandler(ignoreCancelled = true, priority = EventPriority.LOWEST)
-    public void onPlayerShear(PlayerShearEntityEvent event) {
+    public void onPlayerDropItem(PlayerDropItemEvent event) {
         if (listenerService.shouldCancelEvent(event)) {
             event.setCancelled(true);
         }
     }
 
     @EventHandler(ignoreCancelled = true, priority = EventPriority.LOWEST)
-    public void onPlayerFish(PlayerFishEvent event) {
+    public void onPlayerHeldItem(PlayerItemHeldEvent event) {
         if (listenerService.shouldCancelEvent(event)) {
             event.setCancelled(true);
         }
     }
 
     @EventHandler(ignoreCancelled = true, priority = EventPriority.LOWEST)
-    public void onPlayerEditBook(PlayerEditBookEvent event) {
+    public void onPlayerConsumeItem(PlayerItemConsumeEvent event) {
         if (listenerService.shouldCancelEvent(event)) {
             event.setCancelled(true);
         }
     }
 
+    private boolean isInventoryWhitelisted(InventoryView inventory) {
+        if (inventory == null) {
+            return false;
+        }
+        Set<String> whitelist = settings.getProperty(RestrictionSettings.UNRESTRICTED_INVENTORIES);
+        return whitelist.contains(ChatColor.stripColor(inventory.getTitle()));
+    }
+
     @EventHandler(ignoreCancelled = true, priority = EventPriority.LOWEST)
-    public void onPlayerInteractAtEntity(PlayerInteractAtEntityEvent event) {
-        if (listenerService.shouldCancelEvent(event)) {
+    public void onPlayerInventoryOpen(InventoryOpenEvent event) {
+        final HumanEntity player = event.getPlayer();
+
+        if (listenerService.shouldCancelEvent(player)
+            && !isInventoryWhitelisted(event.getView())) {
             event.setCancelled(true);
+
+            /*
+             * @note little hack cause InventoryOpenEvent cannot be cancelled for
+             * real, cause no packet is sent to server by client for the main inv
+             */
+            bukkitService.scheduleSyncDelayedTask(player::closeInventory, 1);
         }
     }
 
+    @EventHandler(ignoreCancelled = true, priority = EventPriority.LOWEST)
+    public void onPlayerInventoryClick(InventoryClickEvent event) {
```

Comment:
Just as feedback as a reviewer—unfortunately, I've had to check out the branch and reorder the methods the old way again so I could get a sensible diff. :) This would have benefited from two commits: one with actual method content changes and then a subsequent one reordering them (or the other way around, doesn't matter)

Reply:
Sorry! xD